modin/data_management/factories/factories.py

```diff
@@ -566,16 +566,13 @@ class ExperimentalPandasOnCloudpythonFactory(ExperimentalRemoteFactory):
     wrapped_factory = PandasOnPythonFactory
 
 
-@doc(_doc_factory_class, backend_name="experimental OmnisciOnRay")
-class ExperimentalOmnisciOnRayFactory(BaseFactory):
+@doc(_doc_factory_class, backend_name="experimental ArrowOnOmnisci")
+class ExperimentalArrowOnOmnisciFactory(BaseFactory):
```

Comment:
don't you think that `OmnisciOnPython` is a more appropriate name for this backend?

Speaking formally, a backend is the content of partitions - for PandasOn* the partitions contain pandas DataFrames, for PyarrowOnRay, partitions contain Pyarrow tables, and so on. An engine is some kind of partition manager - for *OnRay the ray manages partitions, for *OnPython it's only a python role to do this and so on.

In the case of Omnisci backend pseudo-partitions contain Omnisci server, so the proper naming would be OmnisciOn*. Previously the partitions manager appears to be Ray, when we'll remove it, pseudo-partitions would be managed by the Python code only, so it seems that the Python becomes the new engine, and so the resulted name for the "new" backend should be `OmnisciOnPython`. What do you think?

Reply:
That is the place where we need to reach out an agreement. `ArrowOnOmnisci` is one of the choices we discussed internally and probably requires to be changed. I like your thoughts on this but let's collect other opinions and pick one that is the most suitable.

Reply:
cc @modin-project/modin-core , @modin-project/modin-omnisci

Reply:
Having discussed internally, decided to use `OmnisciOnPython`.

Reply:
@modin-project/modin-core , any objections?